code/script.js

```diff
@@ -1,19 +1,172 @@
-// Start here
+function main() {
+  // Step 1 - Welcome and introduction
+  alert(
+    `Welcome to our Stockholm Pizzeria.
+Ready to Start?
+Click 'OK' to begin.`
+  );
 
-// Step 1 - Welcome and introduction
-// Your code goes here
-alert(
-  `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
-)
+  let name = prompt("Please type your name.");
 
-// Step 2 - Food choice
-// Your code goes here
+  while (!name.trim()) {
+    alert("Please enter a valid name.");
+    name = prompt("Please type your name.");
+  }
 
-// Step 3 - Subtype choice
-// Your code goes here
+  alert("Hi, " + name + "!");
 
-// Step 4 - Age
-// Your code goes here
+  // Step 2 - Food choice
+  let userChoice = prompt(
+    `What would you like to order?
+Choose and type the number:
+1-Pizza
+2-Pasta
+3-Salad`
+  );
 
-// Step 5 - Order confirmation
-// Your code goes here
+  let hasCorrectResult = ["1", "2", "3"].includes(userChoice);
+
+  while (!hasCorrectResult) {
+    alert("Invalid answer. Please choose from 1, 2, 3.");
+    userChoice = prompt(
+      `What would you like to order?
+  Choose and type the number:
+  1-Pizza
+  2-Pasta
+  3-Salad`
+    );
+    hasCorrectResult = ["1", "2", "3"].includes(userChoice);
+  }
+
+  let foodTypes = ["Pizza", "Pasta", "Salad"];
+  let chosenFoodType = foodTypes[parseInt(userChoice) - 1];
+
+  alert("You have selected " + chosenFoodType + "!");
+
+  // Step 3 - Subtype choice
+  let subtypeChoice;
+  switch (chosenFoodType) {
+    case "Pizza":
+      subtypeChoice = prompt(
+        `What type of Pizza would you like?
+    Choose and type the number:
+    1 - Margherita
+    2 - Pepperoni
+    3 - Hawaiian`
+      );
+      break;
+
+    case "Pasta":
+      subtypeChoice = prompt(
+        `What type of Pasta would you like?
+    Choose and type the number:
+    1 - Spaghetti Carbonara
+    2 - Lasagna
+    3 - Penne Arrabbiata`
+      );
+      break;
+
+    case "Salad":
+      subtypeChoice = prompt(
+        `What type of Salad would you like?
+    Choose and type the number:
+    1 - Caesar Salad
+    2 - Greek Salad
+    3 - Caprese Salad`
+      );
+      break;
+
+    default:
+      alert("Something went wrong. Please start over.");
```

Comment:
Try to be consistent and use the same behaviour. Earlier you informed the user `alert("Invalid answer. Please choose from 1, 2, 3.");` now instead you are asking to start over. It's better to choose with way you prefer and stick to it :)